### ISSUE TITLE
[FEATURE] Add Specs for Point-in-Time actions

### ISSUE THREAD

**Issue by nhtruong** [opensearch-project/opensearch-api-specification#99]
There are [6 PIT actions](https://opensearch.org/docs/latest/point-in-time-api/) that was added to OS 2.4:
- `POST {index}/_search/point_in_time`
- `GET _search/point_in_time/_all`
- `GET _cat/pit_segments/_all`
- `GET _cat/pit_segments`
- `DELETE _search/point_in_time/_all`
- `DELETE _search/point_in_time`

Follow the [Developer's Guide](./DEVELOPER_GUIDE.md) doc to fill in these actions.
You might have to manually make calls to an OS cluster to figure out the shape of the response bodies.

**Comment by Tokesh** [comment]
Hey! I would like to take this issue! :)

**Comment by nhtruong** [comment]
@Tokesh Split this into 2 issues. The other one can be found here https://github.com/opensearch-project/opensearch-api-specification/issues/100 and it is blocked by #91

**Comment by Tokesh** [comment]
> @Tokesh Split this into 2 issues. The other one can be found here #100 and it is blocked by #91

I am delaying completing this issue due to my exams. But your issue is my priority and I hope I don’t let you down too much! I made a draft pull request - you can see it right now and maybe some errors are already visible right now?

I had two questions while I was writing this part.
The first question is do I need to specify $version = 2? Or which version should I specify?

And the second question, should I write two APIs(```_cat/pit_segments/_all``` and ```_cat/pit_segments```) in one model or divide them into two different ones? I saw in one of your implementations there were two APIs in one model.

**Comment by Tokesh** [comment]
UPD: And here comes the most interesting part! I have passed all the checks in the PR and there are no errors from Smithy. It looks like this is correct, is it possible to check somehow? I mean, is there some kind of guideline I can pick it up and test it or something?

**Comment by Tokesh** [comment]
Hello everyone!
PR is ready for your review! 
I tried to do everything as it is written in the Developer Guide.

**Comment by VachaShah** [comment]
@nhtruong We are also missing delete PIT by PIT ID in https://github.com/opensearch-project/opensearch-api-specification/blob/main/model/_global/delete_all_pits/operations.smithy

**Comment by dblock** [comment]
Closing via https://github.com/opensearch-project/opensearch-api-specification/pull/103.
